daemon/graphdriver/devmapper/deviceset.go

```diff
@@ -553,6 +580,62 @@ func (devices *DeviceSet) migrateOldMetaData() error {
 	return nil
 }
 
+// Cleanup deleted devices. It assumes that all the devices have been
+// loaded in the hash table. Should be called with devices.Lock() held.
+// Will drop the lock for device deletion and return with lock acquired.
+func (devices *DeviceSet) cleanupDeletedDevices() error {
+	var deletedDevices []*devInfo
```

Comment:
Unlike in C in go it's better to decalre vars closer to their usage. So it should be after zero-check.